examples/AdvancedOperations/AddDisplayUploadAd.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Examples\AdvancedOperations;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\V3\GoogleAdsException;
+use Google\Ads\GoogleAds\Util\V1\ResourceNames;
+use Google\Ads\GoogleAds\V3\Common\AdMediaBundleAsset;
+use Google\Ads\GoogleAds\V3\Common\DisplayUploadAdInfo;
+use Google\Ads\GoogleAds\V3\Common\MediaBundleAsset;
+use Google\Ads\GoogleAds\V3\Enums\AdGroupAdStatusEnum\AdGroupAdStatus;
+use Google\Ads\GoogleAds\V3\Enums\AssetTypeEnum\AssetType;
+use Google\Ads\GoogleAds\V3\Enums\DisplayUploadProductTypeEnum\DisplayUploadProductType;
+use Google\Ads\GoogleAds\V3\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V3\Resources\Ad;
+use Google\Ads\GoogleAds\V3\Resources\AdGroupAd;
+use Google\Ads\GoogleAds\V3\Resources\Asset;
+use Google\Ads\GoogleAds\V3\Services\AdGroupAdOperation;
+use Google\Ads\GoogleAds\V3\Services\AssetOperation;
+use Google\Ads\GoogleAds\V3\Services\MutateAdGroupAdsResponse;
+use Google\Ads\GoogleAds\V3\Services\MutateAssetResult;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\BytesValue;
+use Google\Protobuf\StringValue;
+
+/**
+ * This code example adds a display upload ad to a given ad group.
+ * To get ad groups, run GetAdGroups.php.
+ */
+class AddDisplayUploadAd
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    private const AD_GROUP_ID = 'INSERT_AD_GROUP_ID_HERE';
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments(
+            [
```

Comment:
nit: you could remove the extra indentation for `[` and `]`, see how [GetCampaigns.php](https://github.com/googleads/google-ads-php/blob/master/examples/BasicOperations/GetCampaigns.php#L44) does for reference.

I noticed a few other occurrences later in the code. This is PSR-12 compliant so not a problem if you prefer to keep it that way.

Reply:
It's hard to make the code formatter fix in this way.
I'll fix for this time but  will probably need to leave it as is sometimes for faster implementation.